exchange/exchange.go

```diff
@@ -190,6 +191,30 @@ func (e *exchange) HoldAuction(ctx context.Context, r AuctionRequest, debugLog *
 
 	responseDebugAllow, accountDebugAllow, debugLog := getDebugInfo(r.BidRequest, requestExt, r.Account.DebugAllow, debugLog)
 
+	if r.RequestType == metrics.ReqTypeORTB2Web || r.RequestType == metrics.ReqTypeORTB2App {
+		//Extract First party data for auction endpoint only
+		if responseDebugAllow {
+			//ExtractFPDForBidders removes FPD related structures from request.
+			//In order to preserve original request and return it properly in response.ext.debug.resolvedrequest
+			//original request should be copied before FPD execution
+			resolvedBidReq, err := json.Marshal(r.BidRequest)
+			if err != nil {
+				return nil, err
+			}
+			r.ResolvedBidRequest = resolvedBidReq
+		}
+
+		resolvedFPD, fpdErrors := firstpartydata.ExtractFPDForBidders(&openrtb_ext.RequestWrapper{BidRequest: r.BidRequest})
```

Comment:
Are we taking a step back here performance wise by having to wrap the bid request since the wrapper isn't available in `HoldAuction` at the moment? It seems like we could be unmarshalling fields we already unmarshalled. Something to check.